main.go

```diff
@@ -331,9 +334,19 @@ func newApp(configFile string) *cli.App {
 						return err
 					}
 
-					cfg, err := rw.GetActiveContext()
-					if err != nil {
-						return err
+					contextName := c.String(contextNameFlag)
+					var cfg *config.ClusterConfig
+					if contextName != "" {
+						cfg, err = rw.GetContext(contextName)
+						if err != nil {
+							return err
+						}
+
+					} else {
+						cfg, err = rw.GetActiveContext()
+						if err != nil {
+							return err
+						}
 					}
```

Comment:
nit: we can avoid double error handling like this

```suggestion
					contextName := c.String(contextNameFlag)
					if contextName != "" {
						contextName = rw.GetActiveContext()
					} 

					cfg, err := rw.GetContext(contextName)
					if err != nil {
						return err
					}
```